internal/common/constant/constant.go

```diff
@@ -27,20 +27,24 @@ import (
 // Annotations on the other hand should be used to define ownership
 // of the specific Object, such as Backup/Restore.
 const (
-	OadpLabel             = v1alpha1.OadpOperatorLabel
-	OadpLabelValue        = TrueString
-	ManagedByLabel        = "app.kubernetes.io/managed-by"
-	ManagedByLabelValue   = "oadp-nac-controller" // TODO why not use same project name as in PROJECT file?
-	NabOriginNACUUIDLabel = v1alpha1.OadpOperatorLabel + "-nab-origin-nacuuid"
-	NarOriginNACUUIDLabel = v1alpha1.OadpOperatorLabel + "-nar-origin-nacuuid"
-
-	NabOriginNameAnnotation      = v1alpha1.OadpOperatorLabel + "-nab-origin-name"
-	NabOriginNamespaceAnnotation = v1alpha1.OadpOperatorLabel + "-nab-origin-namespace"
-	NarOriginNameAnnotation      = v1alpha1.OadpOperatorLabel + "-nar-origin-name"
-	NarOriginNamespaceAnnotation = v1alpha1.OadpOperatorLabel + "-nar-origin-namespace"
-
-	NabFinalizerName = "nonadminbackup.oadp.openshift.io/finalizer"
-	NarFinalizerName = "nonadminrestore.oadp.openshift.io/finalizer"
+	OadpLabel               = v1alpha1.OadpOperatorLabel
+	OadpLabelValue          = TrueString
+	ManagedByLabel          = "app.kubernetes.io/managed-by"
+	ManagedByLabelValue     = "oadp-nac-controller" // TODO why not use same project name as in PROJECT file?
+	NabOriginNACUUIDLabel   = v1alpha1.OadpOperatorLabel + "-nab-origin-nacuuid"
+	NarOriginNACUUIDLabel   = v1alpha1.OadpOperatorLabel + "-nar-origin-nacuuid"
+	NabslOriginNACUUIDLabel = v1alpha1.OadpOperatorLabel + "-nabsl-origin-nacuuid"
+
+	NabOriginNameAnnotation        = v1alpha1.OadpOperatorLabel + "-nab-origin-name"
+	NabOriginNamespaceAnnotation   = v1alpha1.OadpOperatorLabel + "-nab-origin-namespace"
+	NarOriginNameAnnotation        = v1alpha1.OadpOperatorLabel + "-nar-origin-name"
+	NarOriginNamespaceAnnotation   = v1alpha1.OadpOperatorLabel + "-nar-origin-namespace"
+	NabslOriginNameAnnotation      = v1alpha1.OadpOperatorLabel + "-nabsl-origin-name"
+	NabslOriginNamespaceAnnotation = v1alpha1.OadpOperatorLabel + "-nabsl-origin-namespace"
+
+	NabFinalizerName   = "nonadminbackup.oadp.openshift.io/finalizer"
+	NarFinalizerName   = "nonadminrestore.oadp.openshift.io/finalizer"
+	NabslFinalizerName = "nabsl.oadp.openshift.io/finalizer"
```

Comment:
lets be uniform here, either use shortname for all the CRs or fullname for all of them in finalizer label.

Reply:
Ok changed it to a very long one as I prefer to be explicit ;)
```go
	NabslFinalizerName = "nonadminbackupstoragelocation.oadp.openshift.io/finalizer"
```
will commit soon.

Reply:
Can we close this as resolved? or do you prefer shorter names across all? I am fine either way.